plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java

```diff
@@ -1441,24 +1440,6 @@ public void close()
         }
     }
 
-    // container class for readability. Each value is one for a partitionKey, in order they appear in the schema
-    private static class PartitionValues
-    {
-        private final List<String> values;
-
-        private static PartitionValues make(String... values)
-        {
-            return new PartitionValues(Arrays.asList(values));
-        }
-
-        private PartitionValues(List<String> values)
-        {
-            this.values = ImmutableList.copyOf(requireNonNull(values, "values is null"));
-        }
-
-        public List<String> getValues()
-        {
-            return values;
-        }
+    private record PartitionValues(String... values){
```

Comment:
Is PartitionValues no longer immutable?

Reply:
True. This change would revert the added immutability in the previous commit. I'll wait for the NPE fix from https://github.com/trinodb/trino/pull/14799 to land and use the same logic in the record constructor to make the `values` actually immutable.

Reply:
I merged that one

Reply:
The arrays can be made immutable by converting them to a list.
Changing the code to work with a constructor for `PartitionValues` based on list defies the purpose of  having a concise way to define a variable number of parameters.
It seems to me that the actual design is much better then the one based on `record`.
Closing the PR.